edit/edit.go

```diff
@@ -290,6 +290,45 @@ func where(ed *Editor, a Address) (addr, error) {
 	return at, nil
 }
 
+// WhereLine returns the line number corresponding to the start and
+// end of the address.
+// Dot is set to the address.
+func (ed *Editor) WhereLine(a Address) (lineFrom, lineTo int64, err error) {
+	ed.buf.lock.RLock()
+	defer ed.buf.lock.RUnlock()
+	from, to, err := whereLine(ed, a)
+	return from, to, err
+}
+
+func whereLine(ed *Editor, a Address) (lineFrom, lineTo int64, err error) {
+	at, err := a.addr(ed)
+	if err != nil {
+		return 0, 0, err
+	}
+	var i, to, from int64
+	from = 1 // line numbers are 1 based.
+	for i = 0; i < at.from; i++ {
+		r, err := ed.buf.rune(i)
+		if err != nil {
+			return 0, 0, err
+		} else if r == '\n' {
+			from++
+		}
+	}
+	to = from
+	i++
+	for ; i < at.to; i++ {
+		r, err := ed.buf.rune(i - 1)
```

Comment:
Why don't you want to check the rune at `at.to-1`? That rune is within the address too.
We need to add some more tests:
* where the selection begins with \n
* where the selection ends with \n
* where the selection {begins, ends} just {after,before} \n

I want to make sure that we correctly handle these edge cases.